armory/instrument/config.py

```diff
@@ -0,0 +1,454 @@
+            # Don't include % symbol
+            f_result = f"{np.mean(result):.2}"
+        else:
+            f_result = f"{np.mean(result):.2%}"
+        log.success(
+            f"{name} on {self.task_type} examples w.r.t. {self.wrt} labels: {f_result}"
+        )
+
+
+def _task_metric(
+    name,
+    metric_kwargs,
+    use_mean=True,
+    include_benign=True,
+    include_adversarial=True,
+    include_targeted=True,
+    record_final_only=True,
+):
+    """
+    Return list of meters generated for this specific task
+    """
+    meters = []
+    metric = metrics.get_supported_metric(name)
+    final_kwargs = {}
+    if name in MEAN_AP_METRICS:
+        final_suffix = name
+        final = metrics.MeanAP(metric)
+        final_kwargs = metric_kwargs
+
+        name = f"input_to_{name}"
+        metric = metrics.get_supported_metric("identity_unzip")
+        metric_kwargs = None
+        record_final_only = True
+    elif name == "entailment":
+        final = metrics.get_supported_metric("total_entailment")
+        final_suffix = "total_entailment"
+    elif name == "word_error_rate":
+        final = metrics.get_supported_metric("total_wer")
+        final_suffix = "total_word_error_rate"
+    elif use_mean:
+        final = np.mean
+        final_suffix = f"mean_{name}"
+    else:
+        final = None
+        final_suffix = ""
+        record_final_only = False
+
+    if include_benign:
+        meters.append(
+            Meter(
+                f"benign_{name}",
+                metric,
+                "scenario.y",
+                "scenario.y_pred",
+                metric_kwargs=metric_kwargs,
+                final=final,
+                final_name=f"benign_{final_suffix}",
+                final_kwargs=final_kwargs,
+                record_final_only=record_final_only,
+            )
+        )
+    else:
+        meters.append(None)
+    if include_adversarial:
+        meters.append(
+            Meter(
+                f"adversarial_{name}",
+                metric,
+                "scenario.y",
+                "scenario.y_pred_adv",
+                metric_kwargs=metric_kwargs,
+                final=final,
+                final_name=f"adversarial_{final_suffix}",
+                final_kwargs=final_kwargs,
+                record_final_only=record_final_only,
+            )
+        )
+    else:
+        meters.append(None)
+    if include_targeted:
+        meters.append(
+            Meter(
+                f"targeted_{name}",
+                metric,
+                "scenario.y_target",
+                "scenario.y_pred_adv",
+                metric_kwargs=metric_kwargs,
+                final=final,
+                final_name=f"targeted_{final_suffix}",
+                final_kwargs=final_kwargs,
+                record_final_only=record_final_only,
+            )
+        )
+    else:
+        meters.append(None)
+    return meters
+
+
+def task_metrics(
```

Comment:
I think the code would be a bit more readable if `task_metrics()` was renamed to more concretely state what the function is doing

Reply:
Renamed, albeit verbosely.